plugins/destination/mysql/client/types.go

```diff
@@ -1,55 +1,106 @@
-	if strings.HasPrefix(sqlType, "datetime") {
-		// MySQL permits up to microseconds (6 digits) precision
-		return arrow.FixedWidthTypes.Timestamp_us, nil
+const defaultPrecision = 10
+const defaultScale = 0
+
+func isUnsigned(sqlType string) bool {
+	return strings.Contains(sqlType, "unsigned")
+}
+
+func parseStringValue(str string, defaultValue int32) int32 {
+	val, err := strconv.ParseInt(str, 10, 32)
+	if err != nil {
+		return defaultValue
+	}
+	return int32(val)
+}
+
+func getPrecisionAndScale(dataType string) (precision, scale int32) {
+	str := strings.TrimPrefix(dataType, "decimal")
+	str = strings.TrimPrefix(str, "numeric")
+	if str == "" {
+		return defaultPrecision, defaultScale
 	}
-	sqlTypeToSchemaType := map[string]arrow.DataType{
-		"tinyint(1)":           arrow.FixedWidthTypes.Boolean,
-		"tinyint":              arrow.PrimitiveTypes.Int8,
-		"tinyint(4)":           arrow.PrimitiveTypes.Int8,
-		"smallint":             arrow.PrimitiveTypes.Int16,
-		"smallint(5)":          arrow.PrimitiveTypes.Int16,
-		"smallint(6)":          arrow.PrimitiveTypes.Int16,
-		"smallint(5) unsigned": arrow.PrimitiveTypes.Uint16,
-		"int":                  arrow.PrimitiveTypes.Int32,
-		"int(10) unsigned":     arrow.PrimitiveTypes.Uint32,
-		"int(11)":              arrow.PrimitiveTypes.Int32,
-		"bigint":               arrow.PrimitiveTypes.Int64,
-		"bigint(20)":           arrow.PrimitiveTypes.Int64,
-		"tinyint unsigned":     arrow.PrimitiveTypes.Uint8,
-		"tinyint(3) unsigned":  arrow.PrimitiveTypes.Uint8,
-		"smallint unsigned":    arrow.PrimitiveTypes.Uint16,
-		"int unsigned":         arrow.PrimitiveTypes.Uint32,
-		"bigint unsigned":      arrow.PrimitiveTypes.Uint64,
-		"bigint(20) unsigned":  arrow.PrimitiveTypes.Uint64,
-		"float":                arrow.PrimitiveTypes.Float32,
-		"double":               arrow.PrimitiveTypes.Float64,
-		"binary(16)":           types.ExtensionTypes.UUID,
-		"blob":                 arrow.BinaryTypes.LargeBinary,
-		"text":                 arrow.BinaryTypes.LargeString,
-		"json":                 types.ExtensionTypes.JSON,
+	str = strings.TrimPrefix(str, "(")
+	str = strings.TrimSuffix(str, ")")
+	parts := strings.Split(str, ",")
+
+	switch len(parts) {
+	case 1:
+		precision = parseStringValue(parts[0], defaultPrecision)
+		scale = defaultScale
+	case 2:
+		precision = parseStringValue(parts[0], defaultPrecision)
+		scale = parseStringValue(parts[1], defaultScale)
+	default:
+		precision = defaultPrecision
+		scale = defaultScale
 	}
+	return precision, scale
+}
 
-	if v, ok := sqlTypeToSchemaType[sqlType]; ok {
-		return v, nil
+func mySQLTypeToArrowType(sqlType string) arrow.DataType {
+	if sqlType == "binary(16)" {
+		return types.ExtensionTypes.UUID
+	}
+	if sqlType == "tinyint(1)" {
+		return arrow.FixedWidthTypes.Boolean
+	}
+	if strings.HasPrefix(sqlType, "datetime") {
+		return arrow.FixedWidthTypes.Timestamp_us
+	}
+	if strings.HasPrefix(sqlType, "decimal") || strings.HasPrefix(sqlType, "numeric") {
```

Comment:
It looks like this might fail in the future if we start supporting general array types via Arrow, but right now this is not yet supported on the source side so we're okay. It'll be a bigger change in any case